### PR TITLE
Add transforms to support users with matching names

### PR DESCRIPTION
**Pull request by iwootten**
### What is the context of this PR?

Adds a number of transforms to placeholders for displaying middle names when first and last name match.

Changes to the `test_list_collector_same_name_items` now demonstrate both transforms at work. Full names for all householders will be displayed on the summary page if any householder has duplicate names to another, the edit page will show middle names only if that person shares their first/last name with someone else in the household.

NB. Originally I had used a new source of "list_item_id" to return the list_Item_id, but this seemed very specific and after discussion felt it better we add a "location" source. The renderer already takes both a list_item_id and location and we actually return that of list_item_id in the new transforms - to use location requires significant change to the existing context design (as they don't always have the property) which didn't seem required for this feature. If we ever have a need for other location properties in transforms, then we can perhaps look at refactoring contexts to support them.

### How to review 

Describe the steps required to test the changes (include screenshots if appropriate).

### CODE REVIEW

**Comment by iwootten** [comment]
[[T] Resolve name to be displayed when duplicate names exist (M)](https://trello.com/c/cIBH9hWa/4112-t-resolve-name-to-be-displayed-when-duplicate-names-exist-m)

**Review by ajmaddaford — CHANGES_REQUESTED** [review]

File: test_schemas/en/test_list_collector_same_name_items.json [hunk 1]
```diff
@@ -315,14 +315,36 @@
...
-                                                            "identifier": ["first-name", "last-name"],
-                                                            "source": "answers"
+                                                        "same_name_item_ids": {
+                                                            "source": "list",
+                                                            "id_selector": "same_name_items",
+                                                            "identifier": "people"
+                                                        },
+                                                        "list_item_id": {
+                                                            "source": "location",
+                                                            "identifier": "list_item_id"
```
[the remark this answers is not in the document]
Comment (iwootten) [reply test_schemas/en/test_list_collector_same_name_items.json:327]: I think list will be reserved, so we'd need to use something else `list_to_check`?

File: test_schemas/en/test_list_collector_same_name_items.json [hunk 2]
```diff
@@ -315,14 +315,36 @@
...
                                                         }
-                                                    },
-                                                    "transform": "concatenate_list"
+                                                    }
+                                                },
+                                                {
+                                                    "transform": "format_name",
+                                                    "arguments": {
+                                                        "include_middle_name": { "source": "previous_transform" },
```
Comment (ajmaddaford) [inline test_schemas/en/test_list_collector_same_name_items.json:334]: `include_middle_name` should be `include_middle_names` to align with the `middle_names` property.

**Review by ajmaddaford — APPROVED** [review]

File: scripts/run_validator.sh [hunk 3]
```diff
@@ -1,4 +1,4 @@
 #!/usr/bin/env bash
-tag=latest
+tag=resolve-name-display
```
Comment (MebinAbraham) [inline scripts/run_validator.sh:2]: Reminder to revert before merge.

**Review by MebinAbraham — APPROVED** [review]

**Pull request merged**